src/main/java/com/minecolonies/core/entity/pathfinding/pathjobs/PathJobCanSee.java

```diff
@@ -21,37 +25,39 @@ public class PathJobCanSee extends AbstractPathJob
     private final LivingEntity lookTarget;
 
     /**
-     * The entity to move
+     * The position we want to search around from, usually guarding pos, so guide the heuristic there from the current entity position
      */
-    private final LivingEntity searchingEntity;
+    private final BlockPos searchAroundPos;
 
     public PathJobCanSee(
-      final LivingEntity searchingEntity,
+      final Mob searchingEntity,
       final LivingEntity lookTarget,
       final Level world,
-      @NotNull final BlockPos pos, final int range)
+      @NotNull final BlockPos searchAroundPos, final int range)
     {
-        super(world, searchingEntity.blockPosition(), pos, range, searchingEntity);
+        super(world, PathfindingUtils.prepareStart(searchingEntity), range, new PathResult<PathJobCanSee>(), searchingEntity);
 
-        this.searchingEntity = searchingEntity;
+        this.searchAroundPos = searchAroundPos;
         this.lookTarget = lookTarget;
     }
 
     @Override
     protected double computeHeuristic(final int x, final int y, final int z)
     {
-        return BlockPosUtil.distManhattan(start.getX(), start.getY(), start.getZ(), x, y, z);
+        return BlockPosUtil.distManhattan(searchAroundPos.getX(), searchAroundPos.getY(), searchAroundPos.getZ(), x, y, z);
     }
 
     @Override
     protected boolean isAtDestination(final MNode n)
     {
-        if (end.getY() - n.y > 2)
+        if (start.getY() - n.y > 2)
```

Comment:
this seems weird.

Reply:
ye that is wrong, should be the search around pos (guardpos)